core/src/test/java/org/apache/accumulo/core/client/ZooKeeperInstanceTest.java

```diff
@@ -128,33 +129,33 @@ public void testGetInstanceID_Direct() {
     assertEquals(IID_STRING, zki.getInstanceID());
   }
 
-  @Test(expected = RuntimeException.class)
+  @Test
   public void testGetInstanceID_NoMapping() {
     ClientConfiguration config = createMock(ClientConfiguration.class);
     expect(zc.get(Constants.ZROOT + Constants.ZINSTANCES + "/instance")).andReturn(null);
     replay(zc);
     EasyMock.reset(config, zcf);
-    new ZooKeeperInstance(config, zcf);
+    assertThrows(RuntimeException.class, () -> new ZooKeeperInstance(config, zcf));
```

Comment:
FWIW, the current version of JUnit supports this. This is something that we can easily apply to all our code today, and it would make this migration go more smoothly, at least for the code review portion.

Reply:
Do you think it would be a good idea to make a separate PR and take care of all of these cases first? Might make sense since even if we decide to not upgrade to JUnit 5 for whatever reason, that could still be a working improvement for our JUnit 4 tests.

Reply:
Yes. These would be good to have first. Then, you can rebase this onto that, in order to make this migration smaller and easier to review.

Reply:
addressed in #2435